### ISSUE TITLE
Postgres: persistentVolume mounted with incorrect file permissions

### ISSUE THREAD

**Issue by royklopper** [kubedb/project#697]
Thank you for such a nice piece of software, it makes our devops lives much easier. I'm trying out the `0.13.0-rc.0` release and I seem to find an issue on restarting postgres pods.

Our enviroment consists of:
* Kubernetes 1.15.4-gke.8 with PSP enabled on GKE
* Istio 1.4.0 with SDS and STRICT MTLS
* KubeDB Operator 0.13.0-rc.0

Having Istio & PodSecurityPolicies enabled we first needed to add the `NET_ADMIN` to the `allowedCapabilities` of the Policy.

A typical Postgres deployment looks like:

```yaml
apiVersion: kubedb.com/v1alpha1
kind: Postgres
metadata:
  name: postgres
  namespace: xxx
spec:
  version: 11.2
  replicas: 2
  standbyMode: Hot
  storageType: Durable
  storage:
    storageClassName: ssd
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 64Gi
  terminationPolicy: DoNotTerminate
  podTemplate:
    spec:
      nodeSelector:
        some.tags: to-check
  archiver:
    storage:
      storageSecretName: xxx-credentials
      gcs:
        bucket: xxx-backup-xxx
```

Here it comes; the first time the database is created properly, we could use it for a time. But after a patch from KubeDB, it stops booting and it'll get in Failed state with output of the following message:

```
I1207 16:27:42.738345       1 log.go:172] FLAG: --alsologtostderr="false"
I1207 16:27:42.738388       1 log.go:172] FLAG: --bypass-validating-webhook-xray="false"
I1207 16:27:42.738394       1 log.go:172] FLAG: --enable-analytics="true"
I1207 16:27:42.738399       1 log.go:172] FLAG: --help="false"
I1207 16:27:42.738405       1 log.go:172] FLAG: --log-flush-frequency="5s"
I1207 16:27:42.738411       1 log.go:172] FLAG: --log_backtrace_at=":0"
I1207 16:27:42.738416       1 log.go:172] FLAG: --log_dir=""
I1207 16:27:42.738422       1 log.go:172] FLAG: --logtostderr="true"
I1207 16:27:42.738427       1 log.go:172] FLAG: --stderrthreshold="0"
I1207 16:27:42.738432       1 log.go:172] FLAG: --use-kubeapiserver-fqdn-for-aks="true"
I1207 16:27:42.738437       1 log.go:172] FLAG: --v="3"
I1207 16:27:42.738442       1 log.go:172] FLAG: --vmodule=""
We want "dashboard-postgres-0" as our leader
I1207 16:27:42.875477       1 leaderelection.go:205] attempting to acquire leader lease  production/dashboard-postgres-leader-lock...
Running as Replica
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
Attempting pg_isready on primary
I1207 16:28:00.334908       1 leaderelection.go:214] successfully acquired lease production/dashboard-postgres-leader-lock
Got leadership, now do your jobs
I1207 16:28:00.359373       1 pod.go:56] Patching Pod production/dashboard-postgres-0 with {"metadata":{"labels":{"kubedb.com/role":"primary"}}}
I1207 16:28:00.395678       1 pod.go:56] Patching Pod production/dashboard-postgres-1 with {"metadata":{"labels":{"kubedb.com/role":"replica"}}}
Attempting pg_isready on primary
Postgres promotion trigger_file found. Running primary run script
Running as Primary
pg_ctl: another server might be running; trying to start server anyway
waiting for server to start....2019-12-07 16:28:00.982 GMT [40] LOG:  skipping missing configuration file "/etc/config/user.conf"
2019-12-07 16:28:00.982 GMT [40] LOG:  skipping missing configuration file "/etc/config/user.conf"
2019-12-07 16:28:00.983 UTC [40] FATAL:  data directory "/var/pv/data" has invalid permissions
2019-12-07 16:28:00.983 UTC [40] DETAIL:  Permissions should be u=rwx (0700) or u=rwx,g=rx (0750).
 stopped waiting
pg_ctl: could not start server
Examine the log output.
I1207 16:28:01.075867       1 log.go:172] exit status 1
```

Is there anything I can do to clarify this issue?

**Comment by YpNo** [comment]
Hi,

I've exactly the same issue. Any news/solution about it ?

Regards.

**Comment by sFrenkie** [comment]
I got similar issue with playing local storage and persistent volume claims. 
For me it helped use `securityContext.fsGroup: 70` (70 is postgres)

**Comment by YpNo** [comment]
Up !

**Comment by rtrive** [comment]
Up!

**Comment by gudleik** [comment]
> For me it helped use `securityContext.fsGroup: 70` (70 is postgres)

This worked for me as well. 
Had same permission issue on a fresh installation of community edition v2021.03.11.

**Comment by tamalsaha** [comment]
This issue has been resolved in v2021.03.17 . Please open a new issue if this persists.
